Bottom: 4b35be938af67d5c82484ea12b235ad6d5ce85b6
Top:    b2cb42e723e9d4722ff38dac9b7e31ae8eccd561
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-03-02 17:08:46 +0800

ptr_ring: don't peek producer

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/include/linux/ptr_ring.h b/include/linux/ptr_ring.h
index a9e28a6..d2ce524 100644
--- a/include/linux/ptr_ring.h
+++ b/include/linux/ptr_ring.h
@@ -185,12 +185,14 @@ static inline bool __ptr_ring_empty(struct ptr_ring *r)
 
 static inline int __ptr_ring_peek_queue_len(struct ptr_ring *r)
 {
-	if (__ptr_ring_empty(r))
-		return 0;
-	else if (r->producer > r->consumer)
-		return r->producer - r->consumer;
+	int i = r->consumer + 64;
+	if (i > r->size)
+		i -= r->size;
+
+	if (r->queue[i])
+		return 64;
 	else
-		return r->producer + r->size - r->consumer;
+		return !__ptr_ring_empty(r);
 }
 
 static inline bool ptr_ring_empty(struct ptr_ring *r)
